starknet_py/net/full_node_client.py

```diff
@@ -120,9 +105,16 @@ async def get_transaction(
         )
         return TransactionSchema().load(res, unknown=EXCLUDE)
 
-    async def get_transaciton_by_block_hash_and_index(
+    async def get_transaction_by_block_hash(
```

Comment:
I know we decided to split get_transaction_* to two methods. Do you remember the reason?

Reply:
Actually get_transaction is more complicated. We have get_transaction (by hash) that's in both gateway and rpc client, and rpc client also has get_transaction_by_block_hash(_and_index) and get_transaction_by_block_number(_and_index) that are non existent on gateway client.